storage/api/src/main/java/org/apache/kafka/server/log/remote/storage/RemoteStorageManager.java

```diff
@@ -75,6 +75,9 @@ enum IndexType {
      * <p>
      * Invoker of this API should always send a unique id as part of {@link RemoteLogSegmentMetadata#remoteLogSegmentId()}
      * even when it retries to invoke this method for the same log segment data.
+     * <p>
+     * This operation is expected to be idempotent. If a copy operation is retried and there is existing content already written,
```

Comment:
Newly added statement gives more clarity on idempotency. 

The given unique id sent as part of `RemoteLogSegmentMetadata` can be used to avoid overwrites in RSM implementation. Caller makes sure it sends unique id for multiple invocations.